apps/api/v2/src/filters/prisma-exception.filter.ts

```diff
@@ -0,0 +1,41 @@
+import { ArgumentsHost, Catch, ExceptionFilter, HttpStatus, Logger } from "@nestjs/common";
+import {
+  PrismaClientInitializationError,
+  PrismaClientKnownRequestError,
+  PrismaClientRustPanicError,
+  PrismaClientUnknownRequestError,
+  PrismaClientValidationError,
+} from "@prisma/client/runtime/library";
+import { Response } from "express";
+
+type PrismaError =
+  | PrismaClientInitializationError
+  | PrismaClientKnownRequestError
+  | PrismaClientRustPanicError
+  | PrismaClientUnknownRequestError
+  | PrismaClientValidationError;
+
+@Catch(
+  PrismaClientInitializationError,
+  PrismaClientKnownRequestError,
+  PrismaClientRustPanicError,
+  PrismaClientUnknownRequestError,
+  PrismaClientValidationError
+)
+export class PrismaExceptionFilter implements ExceptionFilter {
+  private readonly logger = new Logger("PrismaExceptionFilter");
+
+  catch(error: PrismaError, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+
+    this.logger.error(`PrismaError: ${error.message}`, {
+      error,
+    });
+
+    response.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+      success: false,
+      message: "There was an error, please try again later.",
+    });
+  }
```

Comment:
looks good 👍 , what do you think of have few checks on the most common PrismaKnownException codes to have custom error message, like not found,etc...

Reply:
https://www.prisma.io/docs/reference/api-reference/error-reference#error-codes

Reply:
Will do in a different pr.